plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/githubPullRequest.ts

```diff
@@ -0,0 +1,231 @@
+    id: 'publish:github:pull-request',
+    schema: {
+      input: {
+        required: ['owner', 'repo', 'title', 'description', 'branchName'],
+        type: 'object',
+        properties: {
+          owner: {
+            type: 'string',
+            title: 'Repository owner',
+            description: 'The owner of the target repository',
+          },
+          repo: {
+            type: 'string',
+            title: 'Repository',
+            description: 'The github repository to create the file in',
+          },
+          branchName: {
+            type: 'string',
+            title: 'Branch Name',
+            description: 'The name for the branch',
+          },
+          title: {
+            type: 'string',
+            title: 'Pull Request Name',
+            description: 'The name for the pull request',
+          },
+          description: {
+            type: 'string',
+            title: 'Pull Request Description',
+            description: 'The description of the pull request',
+          },
+          sourcePath: {
+            type: 'string',
+            title: 'Working Subdirectory',
+            description:
+              'Subdirectory of working directory to copy changes from',
+          },
+          targetPath: {
+            type: 'string',
+            title: 'Repository Subdirectory',
+            description: 'Subdirectory of repository to apply changes to',
+          },
+        },
+      },
+      output: {
+        required: ['remoteUrl'],
+        type: 'object',
+        properties: {
+          remoteUrl: {
+            type: 'string',
+            title: 'Pull Request URL',
+            description: 'Link to the pull request in Github',
+          },
+        },
+      },
+    },
+    async handler(ctx) {
+      const {
+        owner,
+        repo,
+        branchName,
+        title,
+        description,
+        targetPath,
+        sourcePath,
+      } = ctx.input;
+      const client = await clientFactory(integrations, ctx.input);
+      const fileRoot = sourcePath
+        ? path.join(ctx.workspacePath, sourcePath)
+        : ctx.workspacePath;
+      const localFilePaths = await globby(`${fileRoot}/**/*.*`);
+
+      const fileContents = await Promise.all(
+        localFilePaths.map(p => readFile(p)),
+      );
+
+      const repoFilePaths = localFilePaths.map(p => {
+        const relativePath = path.relative(fileRoot, p);
+        return targetPath ? `${targetPath}/${relativePath}` : relativePath;
+      });
+
+      const changes = [
+        {
+          files: zipObject(
+            repoFilePaths,
+            fileContents.map(buf => buf.toString()),
+          ),
+          commit: title,
+        },
+      ];
+
+      const response = await client.createPullRequest({
+        owner,
+        repo,
+        title,
+        changes,
+        body: description,
+        head: branchName,
+      });
+
+      if (!response) throw new Error('null response from Github');
```

Comment:
Maybe use some errors from `@backstage/errors` package here to throw instead.

Reply:
I created a custom error subclass.